src/mlpack/core/data/split_data.hpp

```diff
@@ -224,6 +228,122 @@ Split(const arma::Mat<T>& input,
+    order = arma::shuffle(order);
+
+  if (trainSize > 0)
+  {
+    trainLabels.set_size(1, trainSize);
+
+    for (size_t i = 0; i < trainSize; i++)
+       trainData[i] = input(0, order(i));
+
+    // Field type has fixed size so we can't use span and assignment
+    // operator.
+    for (size_t i = 0; i < trainSize; i++)
+      trainLabels(0, i) = inputLabel[i];
+  }
+
+  if (testSize <= input.n_cols)
+  {
+    for (size_t i = trainSize; i < input.n_cols - 1; i++)
+       testData[i - trainSize] = input(0,order(i));
+    
+    testLabels.set_size(1, testSize);
+    for (size_t i = trainSize; i < input.n_cols; i++)
+      testLabels(0, i - trainSize) = inputLabel[i];
+  }
+}
+
+template <class fieldType,
+          class = std::enable_if_t<
+              arma::is_Col<typename fieldType::object_type>::value ||
+              arma::is_Mat_only<typename fieldType::object_type>::value>>
+void Split(fieldType& input, 
+           fieldType& trainData,
+           fieldType& testData, 
+           const double testRatio, 
+           const bool shuffleData = true) 
+{
+  const size_t testSize = static_cast<size_t>(input.n_cols * testRatio);
+  const size_t trainSize = input.n_cols - testSize;
+
+  arma::uvec order = arma::linspace<arma::uvec>(0, input.n_cols - 1,
+      input.n_cols);
+  if (shuffleData)
+    order = arma::shuffle(order);
+
+  if (trainSize > 0)
+  {
+    for (size_t i = 0; i < trainSize; i++)
+       trainData[i] = input(0, order(i));
+  }
+
+  if (testSize <= input.n_cols)
+  {
+    for (size_t i = trainSize; i < input.n_cols - 1; i++)
+       testData[i - trainSize] = input(0,order(i));
+  }
+}
+
+template <class fieldType,
+          class = std::enable_if_t<
+              arma::is_Col<typename fieldType::object_type>::value ||
+              arma::is_Mat_only<typename fieldType::object_type>::value>>
+std::tuple<fieldType, fieldType, fieldType, fieldType>
+Split(fieldType& input, 
+      fieldType& inputLabel,
+      const double testRatio,
+      const bool shuffleData = true)
+{
+  fieldType trainData;
+  fieldType testData;
+  fieldType trainLabel;
+  fieldType testLabel;
+
+  Split(input, inputLabel, trainData, testData, trainLabel, testLabel,
+    testRatio, shuffleData);
+
+  return std::make_tuple(std::move(trainData),
+                         std::move(testData),
+                         std::move(trainLabel),
+                         std::move(testLabel));
+}
+
+template <class fieldType,
+          class = std::enable_if_t<
+              arma::is_Col<typename fieldType::object_type>::value ||
+              arma::is_Mat_only<typename fieldType::object_type>::value>>
+std::tuple<fieldType, fieldType>
+Split(const fieldType& input,
+      const double testRatio,
+      const bool shuffleData = true)
+{
+  fieldType trainData;
+  fieldType testData;
+  Split(input, trainData, testData, testRatio, shuffleData);
+
+  return std::make_tuple(std::move(trainData),
+                         std::move(testData));
+}
```

Comment:
```suggestion
}

```
Probably useful to place an extra line here before closing the namespace. :+1: